Model/Api/OrderManagement.php

```diff
@@ -133,10 +133,14 @@ public function manage(
 
             $this->logHelper->addInfoLog($this->request->getContent());
 
-            $orderStoreId = $this->orderHelper->getOrderStoreIdByDisplayId($display_id);
-            $this->hookHelper->setMagentoStoreId($orderStoreId);
+            // get the store id. try fetching from quote first,
+            // otherwise load it from order if there's one created
+            $storeId = $this->orderHelper->getStoreIdByQuoteId($order)
```

Comment:
Seems correct. However, since the OrderManagement.php is the only place it's used and Helper\Order.php is already loaded I tried to avoid injecting Helper\Cart.php just for that.
I'd leave it like this for now.